storage/google/cloud/storage/bucket.py

```diff
@@ -899,21 +899,39 @@ def lifecycle_rules(self, rules):
         See https://cloud.google.com/storage/docs/lifecycle and
              https://cloud.google.com/storage/docs/json_api/v1/buckets
 
-        :type entries: list of dictionaries
-        :param entries: A sequence of mappings describing each lifecycle rule.
+        :type rules: list of dictionaries
+        :param rules: A sequence of mappings describing each lifecycle rule.
         """
         self._patch_property('lifecycle', {'rule': rules})
 
-    location = _scalar_property('location')
-    """Retrieve location configured for this bucket.
+    _location = _scalar_property('location')
 
-    See https://cloud.google.com/storage/docs/json_api/v1/buckets and
-    https://cloud.google.com/storage/docs/bucket-locations
+    @property
+    def location(self):
+        """Retrieve location configured for this bucket.
 
-    If the property is not set locally, returns ``None``.
+        See https://cloud.google.com/storage/docs/json_api/v1/buckets and
+        https://cloud.google.com/storage/docs/bucket-locations
 
-    :rtype: str or ``NoneType``
-    """
+        If the property is not set locally, returns ``None``.
+
+        :rtype: str or ``NoneType``
+        """
+        return self._location
+
+    @location.setter
```

Comment:
`location` is stored in the `_properties` dict:  we should likely block assignment to it if that value is present.

Hmmmm, wouldn't it be better to make it a keyword argument to `Bucket.create` (since that is the only time it can logically be set by the user)?  Maybe for `client.create_bucket` too.  We could then leave `Bucket.location` a read-only property.

Reply:
We'd still break users if they're potentially dependent on this flow. We could add it to `create_bucket` and `create`, but changing `location` will have the same impact on users as the existing flow (comment above) is the only way to define the location for a new bucket. (Unless I missed another flow).

Reply:
Woah, this got away from me. @crwilcox apologies, can you move forward with @tseaver suggestion?

Reply:
@frankyn holy frankenbug!

We can do this. I may suggest we keep `Client.create_bucket` simple (the docs already point users to use `Bucket.create`. This is the current use pattern.

```
b = Bucket(...)
b.location = ...
b.create()
```

We can add args to `Bucket.create` that will support setting location.

Reply:
Okay, so potentially close this PR (?), because adding location to Bucket.create then gets confusing.

Reply:
That may be the route forward. When I make the PR to replace this one I can close this one. But yes, we need to support setting location, possibly just on Bucket() at creation time.